File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public int solution(int n) {
        int answer = 0;
        ArrayList<Integer> list = new ArrayList<>();
        
        while(n >= 3) {
            list.add(n % 3);
            n /= 3;
        }
        list.add(n);
        
        int k = list.size();
        for(int i = 0; i < list.size(); i++) {
            answer += (int)(Math.pow(3, k - 1)) * list.get(i);
            k--;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n : int):
    answer = 0

    ternary = ''

    while True:
        if n <= 2:
            ternary += str(n)
            break

        ternary += str(n % 3)   

        n = (int)(n / 3)

    answer = int(ternary, 3)

    return answer